src/nvim/highlight_group.c

```diff
@@ -231,90 +223,67 @@ static const char *highlight_init_both[] = {
-
-  "default link @constructor   Special",
-  "default link @operator      Operator",
-
-  "default link @keyword            Keyword",
-  "default link @keyword.function   Statement",
-  "default link @keyword.operator   Operator",
-  "default link @keyword.import     Include",
-  "default link @keyword.storage    StorageClass",
-  "default link @keyword.repeat     Repeat",
-  "default link @keyword.debug      Debug",
-  "default link @keyword.exception  Exception",
-
-  "default link @keyword.conditional    Conditional",
-
-  "default link @keyword.directive         Preproc",
-  "default link @keyword.directive.define  Define",
-
-  "default link @punctuation.delimiter  Delimiter",
-  "default link @punctuation.bracket    Delimiter",
-  "default link @punctuation.special    Special",
-
-  "default link @comment   Comment",
-
-  "default link @comment.error    DiagnosticError",
-  "default link @comment.warning  DiagnosticWarn",
-  "default link @comment.note     DiagnosticInfo",
-  "default link @comment.todo     Todo",
-
-  "@markup.strong         gui=bold           cterm=bold",
-  "@markup.italic         gui=italic         cterm=italic",
-  "@markup.strikethrough  gui=strikethrough, cterm=strikethrough",
-  "@markup.underline      gui=underline,     cterm=underline",
-
-  "default link @markup                 Special",  // fallback for subgroups; never used itself
-  "default link @markup.heading         Title",
-  "default link @markup.environment     Structure",
-  "default link @markup.link            Underlined",
-  "default link @markup.list.checked    DiagnosticOk",
-  "default link @markup.list.unchecked  DiagnosticWarn",
-
-  "default link @diff.plus   Added",
-  "default link @diff.minus  Removed",
-  "default link @diff.delta  Changed",
-
-  "default link @tag            Tag",
-  "default link @tag.delimiter  Delimiter",
+  "default link @variable.builtin Special",
+
+  "default link @constant         Constant",
+  "default link @constant.builtin Special",
+
+  "default link @module         Structure",
+  "default link @module.builtin Special",
+  "default link @label          Label",
+
+  "default link @string             String",
+  "default link @string.regexp      @string.special",
+  "default link @string.escape      @string.special",
+  "default link @string.special     SpecialChar",
+  "default link @string.special.url Underlined",
+
+  "default link @character         Character",
+  "default link @character.special SpecialChar",
+
+  "default link @boolean      Boolean",
+  "default link @number       Number",
+  "default link @number.float Float",
+
+  "default link @type         Type",
+  "default link @type.builtin Special",
+
+  "default link @attribute Macro",
+  "default link @property  Identifier",
+
+  "default link @function         Function",
+  "default link @function.builtin Special",
+
+  "default link @constructor Special",
+  "default link @operator    Operator",
+
+  "default link @keyword Keyword",
```

Comment:
I do think there's an expectation that the different types of keywords are highlighted accordingly; otherwise it's a notable regression over the legacy highlighting (with the same theme).

(In general I agree with the strategy to rely where possible on the implicit fallback; that's what it's there for.)

Reply:
Unfortunately, there are semantic issues here after renaming.
Previously several groups which are now considered (based on naming) under the "keyword"/"statement" umbrella were not there. Those are:
- `@include` / `Include` (no highlighting).
- `@storageclass` / `StorageClass` (no highlighting).
- `@debug` / `Debug` (linked to `Special`).
- `@preproc` / `Preproc` (no highlighting).
- `@define` / `Define` (no highlighting).

With this PR, all their `@keyword.xxx` alternatives will fall back to `@keyword` which is bold. So this is not really a regression, but indeed a change with more groups being recognized (in default color scheme) as keywords.

Reply:
If them being bold is intended, that is fine with me. As I said, you're now taking ownership of the treesitter groups, and colorschemes are on their own to define them.

So this _is_ a regression or change (things that have no highlighting or `Special` highlighting with legacy syntax have bold highlighting with treesitter). The question is whether this is intended/acceptable.